src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs

```diff
@@ -1,184 +1,128 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
 using System.Diagnostics;
-using System.IO;
 using System.Runtime.CompilerServices;
-using System.Threading;
+using System.Text.Internal;
+using System.Text.Unicode;
 
 namespace System.Text.Encodings.Web
 {
-    /// <summary>
-    /// A class which can perform HTML encoding given an allow list of characters which
-    /// can be represented unencoded.
-    /// </summary>
-    /// <remarks>
-    /// Instances of this type will always encode a certain set of characters (such as &lt;
-    /// and &gt;), even if the filter provided in the constructor allows such characters.
-    /// Once constructed, instances of this class are thread-safe for multiple callers.
-    /// </remarks>
-    public unsafe sealed class HtmlEncoder : IHtmlEncoder
+    public abstract class HtmlEncoder : TextEncoder
     {
-        // The default HtmlEncoder (Basic Latin), instantiated on demand
-        private static HtmlEncoder _defaultEncoder;
-
-        // The inner encoder, responsible for the actual encoding routines
-        private readonly HtmlUnicodeEncoder _innerUnicodeEncoder;
-
-        /// <summary>
-        /// Instantiates an encoder using <see cref="UnicodeRanges.BasicLatin"/> as its allow list.
-        /// Any character not in the <see cref="UnicodeRanges.BasicLatin"/> range will be escaped.
-        /// </summary>
-        public HtmlEncoder()
-            : this(HtmlUnicodeEncoder.BasicLatin)
+        public static HtmlEncoder Default
         {
+            get { return DefaultHtmlEncoder.Singleton; }
         }
-
-        /// <summary>
-        /// Instantiates an encoder specifying which Unicode character ranges are allowed to
-        /// pass through the encoder unescaped. Any character not in the set of ranges specified
-        /// by <paramref name="allowedRanges"/> will be escaped.
-        /// </summary>
-        public HtmlEncoder(params UnicodeRange[] allowedRanges)
-            : this(new HtmlUnicodeEncoder(new CodePointFilter(allowedRanges)))
+        public static HtmlEncoder Create(CodePointFilter filter)
         {
+            return new DefaultHtmlEncoder(filter);
         }
-
-        /// <summary>
-        /// Instantiates an encoder using a custom code point filter. Any character not in the
-        /// set returned by <paramref name="filter"/>'s <see cref="ICodePointFilter.GetAllowedCodePoints"/>
-        /// method will be escaped.
-        /// </summary>
-        public HtmlEncoder(ICodePointFilter filter)
-            : this(new HtmlUnicodeEncoder(CodePointFilter.Wrap(filter)))
+        public static HtmlEncoder Create(params UnicodeRange[] allowedRanges)
         {
+            return new DefaultHtmlEncoder(allowedRanges);
         }
+    }
+
+    public sealed class DefaultHtmlEncoder : HtmlEncoder
```

Comment:
For aspnet/EF we always do one top-level type per file to make it easier to navigate the source tree. In VS it doesn't matter so much, but navigating in GitHub is easy if the filename matches the type. While in GitHub you can press <kbd>t</kbd> and type part of the filename to quickly jump to a file in the repo.